packages/dialog/src/index.tsx

```diff
@@ -75,13 +76,21 @@ const DialogInner = forwardRef<HTMLDivElement, DialogProps>(
     const mouseDownTarget = useRef<EventTarget | null>(null);
     const overlayNode = useRef<HTMLDivElement | null>(null);
     const ref = useForkedRef(overlayNode, forwardedRef);
+    const willReturnFocus =
+      returnFocusRef && returnFocusRef.current ? false : true;
 
     const activateFocusLock = useCallback(() => {
       if (initialFocusRef && initialFocusRef.current) {
         initialFocusRef.current.focus();
       }
     }, [initialFocusRef]);
 
+    const deactivateFocusLock = useCallback(() => {
+      if (returnFocusRef && returnFocusRef.current) {
+        setTimeout(() => returnFocusRef.current.focus(), 0);
```

Comment:
It may be worth leaving a comment on why the zero-timeout is needed here. What do you think about either adding a comment like "Without the zero-timeout, focus will likely remain on the button/control" or [linking to the FocusLock docs directly][1]? I like the idea of linking to the docs more.

[1]: https://github.com/theKashey/react-focus-lock#unmounting-and-focus-management